### PR TITLE
[Test] Add test to validate rebalance with locks on files

### PR DESCRIPTION
**Pull request by pranavprakash20**
Test Steps:

1. Create a distributed or distributed-replicate volume and
   populate some data
2. Hold exclusive lock on a file (can use flock)
3. Add-brick and rebalance (make sure this file gets migrated)
4. Again from another program try to hold exclusive lock on this file

Signed-off-by: Pranav <prprakas@redhat.com>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
pytest: -v functional/dht/test_verify_rebalance_with_locks_on_files.py
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.7.3, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/prprakas/pranav/glusto-tests/tests
collected 2 items                                                                                                                                                                            

functional/dht/test_verify_rebalance_with_locks_on_files.py::TestVerifyRebalanceWithFileLock_cplex_distributed_glusterfs::test_verify_rebalance_with_file_lock PASSED                  [ 50%]
functional/dht/test_verify_rebalance_with_locks_on_files.py::TestVerifyRebalanceWithFileLock_cplex_distributed-replicated_glusterfs::test_verify_rebalance_with_file_lock  [ 100%]

**Comment by pranavprakash20** [comment]
@kshithijiyer @aloganat @leelavg @srijan-sivakumar Please review

**Review by kshithijiyer — APPROVED** [review]
LGTM

**Review by srijan-sivakumar — APPROVED** [review]
LGTM

**Review by balakreddy — APPROVED** [review]

File: tests/functional/dht/test_verify_rebalance_with_locks_on_files.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                            self.all_servers_info)
+        self.assertTrue(ret, "Failed to expand the volume")
+
+        # Trigger rebalance
+        ret, _, _ = rebalance_start(self.mnode, self.volname)
+        self.assertEqual(ret, 0, "Failed to start rebalance on the volume %s"
+                         % self.volname)
+
+        # Wait for rebalance to complete
+        ret = wait_for_rebalance_to_complete(self.mnode, self.volname,
```
[the remark this answers is not in the document]
Comment (pranavprakash20) [reply tests/functional/dht/test_verify_rebalance_with_locks_on_files.py:81]: should that be considered here as we are trying the lock from client side?

[the remark this answers is not in the document]
Comment (leelavg) [reply tests/functional/dht/test_verify_rebalance_with_locks_on_files.py:81]: I read 3 step (.. make sure ..) and got this doubt

[the remark this answers is not in the document]
Comment (pranavprakash20) [reply tests/functional/dht/test_verify_rebalance_with_locks_on_files.py:81]: ACK. The file migration has to be validated. Will add the steps for the same

**Review by aloganat — APPROVED** [review]

**Pull request merged**